src/pcap/bosh.go

```diff
@@ -1,55 +1,512 @@
 	}
 
-	if request.Deployment == "" {
+	if boshRequest.Deployment == "" {
 		return fmt.Errorf("invalid message: deployment: %w", errEmptyField)
 	}
 
-	if len(request.Groups) == 0 {
+	if boshRequest.Environment == "" {
+		return fmt.Errorf("invalid message: environment: %w", errEmptyField)
+	}
+
+	if len(boshRequest.Groups) == 0 {
 		return fmt.Errorf("invalid message: instance group(s): %w", errEmptyField)
 	}
 
 	return nil
 }
+
+func (br *BoshResolver) setup() error {
+	br.logger.Debug("setting up BoshResolver", zap.Any("resolver-config", br.Config))
+
+	var tlsConfig *tls.Config
+
+	if br.Config.MTLS != nil {
+		tlsConfig = &tls.Config{
+			MinVersion: tls.VersionTLS12,
+			MaxVersion: tls.VersionTLS13,
+			ClientAuth: tls.RequireAndVerifyClientCert,
+			RootCAs:    br.boshRootCAs,
+		}
+	}
+
+	br.client = &http.Client{
+		Transport: &http.Transport{
+			DialContext: (&net.Dialer{
+				Timeout: 500 * time.Millisecond, //nolint:gomnd // Default configuration
+			}).DialContext,
+			TLSHandshakeTimeout:   500 * time.Millisecond, //nolint:gomnd // Default configuration
+			ResponseHeaderTimeout: 500 * time.Millisecond, //nolint:gomnd // Default configuration
+			ExpectContinueTimeout: 500 * time.Millisecond, //nolint:gomnd // Default configuration
+			DisableKeepAlives:     true,
+			MaxIdleConnsPerHost:   -1,
+			TLSClientConfig:       tlsConfig,
+		},
+		Timeout: time.Second,
+	}
+
+	br.logger.Debug("discovering bosh-UAA endpoint", zap.String("bosh-director", br.DirectorURL.String()))
+	apiResponse, err := br.info()
+	if err != nil {
+		return err
+	}
+
+	br.UaaURLs = apiResponse.UserAuthentication.Options.URLs
+	br.logger.Info("connected to bosh-director", zap.Any("bosh-director", br.DirectorURL.String()))
+	return nil
+}
+
+// info retrieves the BOSH director /info endpoint.
+//
+// Used for startup and health check.
+func (br *BoshResolver) info() (*BoshInfo, error) {
+	// FIXME: Workaround for URL.JoinPath, which is buggy: https://github.com/golang/go/issues/58605
+	infoEndpoint := *br.DirectorURL
+	infoEndpoint.Path = "/info"
+
+	response, err := br.client.Do(&http.Request{
+		Method: http.MethodGet,
+		URL:    &infoEndpoint,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch Bosh Director API from %v: %w", br.Config.RawDirectorURL, err)
+	}
+
+	defer CloseQuietly(response.Body)
+
+	if response.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("received non-OK response from Bosh Director: %s", response.Status)
+	}
+
+	var apiResponse *BoshInfo
+	data, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("could not read bosh-director API response: %w", err)
+	}
+	err = json.Unmarshal(data, &apiResponse)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse bosh-director API response: %w", err)
+	}
+	return apiResponse, nil
+}
+
+// Healthy returns true if the resolver ran setup() and can connect to the BOSH director.
+func (br *BoshResolver) Healthy() bool {
+	if br.client == nil {
```

Comment:
As discussed, panicking in case setup was not called (which requires you to instantiate your own `BoshResolver` and not call `NewBoshResolver`) is OK behaviour. So I will remove these checks.